parquet/write.go

```diff
@@ -54,8 +54,12 @@ func convertSchema(sch *arrow.Schema) *arrow.Schema {
 		case arrow.TypeEqual(f.Type, arrow.ListOf(types.NewUUIDType())),
 			arrow.TypeEqual(f.Type, arrow.ListOf(types.NewInetType())),
 			arrow.TypeEqual(f.Type, arrow.ListOf(types.NewJSONType())),
-			arrow.TypeEqual(f.Type, arrow.ListOf(types.NewMacType())):
+			arrow.TypeEqual(f.Type, arrow.ListOf(types.NewMACType())):
```

Comment:
seems there is issue here with the casting logic. what about `listOf(listOf(string))` or `structOf(listOf(Strings))`. This should be recursive instead of looking for specific types.

Reply:
They will stay as what they are? We don't have a problem with lists/strings/structs. Refer to the `copy(fields, oldFields)` line above the `range` loop.